storage/libradb/src/ledger_store/mod.rs

```diff
@@ -82,18 +82,31 @@ impl LedgerStore {
         Ok(epoch)
     }
 
-    /// Return the ledger infos with their least 2f+1 signatures starting from `start_epoch` to
-    /// the most recent one.
-    /// Note: ledger infos and signatures are only available at the last version of each earlier
-    /// epoch and at the latest version of current epoch.
-    pub fn get_latest_ledger_infos_per_epoch(
+    /// Return the ledger infos reflecting epoch bumps with their least 2f+1 signatures starting
+    /// from `start_epoch` to the most recent one.
+    /// Note: ledger infos and signatures are available at the last version of each earlier epoch
+    /// and at the latest version of current epoch, we filter out the latter.
+    pub fn get_epoch_change_ledger_infos(
         &self,
         start_epoch: u64,
+        ledger_version: Version,
     ) -> Result<Vec<LedgerInfoWithSignatures>> {
         let mut iter = self.db.iter::<LedgerInfoSchema>(ReadOptions::default())?;
         iter.seek(&start_epoch)?;
-        let iter = iter.take(MAX_NUM_EPOCH_CHANGE_LEDGER_INFO + 1);
+        let iter = iter
+            .take(MAX_NUM_EPOCH_CHANGE_LEDGER_INFO + 1)
+            .take_while(|res| {
+                res.as_ref()
+                    .ok()
+                    .map(|(_, x)| {
+                        x.ledger_info().version() <= ledger_version
+                            && x.ledger_info().next_validator_set().is_some()
+                    })
+                    .unwrap_or(false)
```

Comment:
If `res` is an error, this would cause the scan to stop early and the code below to return a shorter result, instead of returning an error.

Reply:
Gooooood catch!